### PR TITLE
Add unit and e2e tests for hierarchy widget

### PR DESCRIPTION
**Pull request by arbenhaziri**
JIRA: https://cdap.atlassian.net/browse/CDAP-17749

### CODE REVIEW

File: cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    cy.get(`${row1Selector} ${dataCy('autocomplete-input')}`).type('column2');
+    cy.get(`${dataCy('option-column2')}`).click();
+    cy.get(`${dataCy('option-column4')}`).click();
+    cy.get(`${dataCy('option-column8')}`).click();
+  });
+  it('Should get the schema', () => {
+    cy.get('[data-cy=plugin-undefined] > [data-cy=widget-wrapper-container] > .WidgetWrapperView-widgetContainer-607 > .abstract-widget-wrapper > div').click()
+    cy.get(`${dataCy('get-schema-btn')}`).click();
+  });
+});
```
Comment (elfenheart) [inline cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts]: nit: newline at EOF

**Review by elfenheart — APPROVED** [review]
minor nit, otherwise LGTM

File: cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts (hunk 1 above)
Comment (arbenhaziri) [reply cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts]: -Added newline at EOF.
-Commits squashed.

File: cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  };
+
+  before(() => {
+    loginIfRequired().then(() => {
+      cy.getCookie('CDAP_Auth_Token').then((cookie) => {
+        if (!cookie) {
+          return;
+        }
+        headers = {
+          Authorization: 'Bearer ' + cookie.value,
```
Comment (antony-britto) [inline cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts]: 'Bearer ${cookie.value}'

File: cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    cy.visit('/pipelines/ns/default/studio');
+    cy.add_node_to_canvas(fileSource)
+    cy.open_transform_panel();
+    cy.add_node_to_canvas(createRecord);
+    cy.pipeline_clean_up_graph_control();
+    cy.fit_pipeline_to_screen();
+    cy.connect_two_nodes(fileSourceId, createRecordId, getGenericEndpoint);
+    cy.open_node_property(fileSourceId);
+  })
+  it('Should add schema with simple files', () => {
```
Comment (antony-britto) [inline cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts:64]: new line is missing for some it() blocks

Comment (arbenhaziri) [reply cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts:64]: PR was updated

File: cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts (hunk 2 above)
Comment (arbenhaziri) [reply cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts]: PR updated

**Review by antony-britto — APPROVED** [review]

File: cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts (hunk 3 above)
Comment (arbenhaziri) [reply cdap-ui/cypress/integration/pipeline.hierarchy.spec.ts:64]: commits are squashed

**Pull request merged**